app/controllers/IndexController.scala

```diff
@@ -121,18 +123,34 @@ class IndexController extends BaseController {
     Ok(scala.xml.XML.loadString(new String(canonicalizer.canonicalizeSubtree(soapEnvelope))))
   }
 
-  def respondToSentry() = Action { implicit request =>
-    val id = request.body.asFormUrlEncoded.get("token").head
+  def sentryLookup(requestType: String, memberFilter: Member => Boolean) = {
     // wtf, Adam
-    val members = (fakeMemberService.getStaff++fakeMemberService.getStudents).filter(m => m.universityId == id)
+    val members = (fakeMemberService.getStaff++fakeMemberService.getStudents).filter(memberFilter)
 
     if(members.isEmpty) {
-      Ok("returnType=51")
+      Ok("returnType=5" + requestType)
     }
     else {
       val response = fakeMemberService.getResponseFor(members.head)
-      val attributes = AttributeConverter.toAttributes(response)
-      Ok("returnType=1\nid=" + members.head.universityId + "\n" + attributes.map(_.productIterator.mkString("=")).mkString("\n"))
+      val attributes = AttributeConverter.toAttributes(response, false)
+      Ok("returnType=" + requestType + "\nid=" + members.head.universityId + "\n" + attributes.map(_.productIterator.mkString("=")).mkString("\n"))
+    }
+  }
+
+  def respondToSentry(requestType: Int, user: Option[String]) = Action { implicit request =>
+    val formData = request.body.asFormUrlEncoded;
+    
+    requestType match {
+      case 1 if (request.method == "POST" && formData.get("token").nonEmpty) =>
```

Comment:
Code style thing, can we remove the brackets after the if on these? Not needed for case match filters, it's effectively a .withFilter